File: EntitiesCs/Customer.cs
using System;
using System.Collections.Generic;

namespace EntitiesCs
{
    public class Customer
    {
        public Customer()
        {
            Nationality = "Argentino";
        }
        public Customer(string Name, int id) : this ()
        {
            this.Name = Name;
            this.Id = id;
        }
        public Customer(string name, int id, DateTime BirthDate) : this(name, id)
        {
            this.BirthDay = BirthDate;
        }
        private string name;
        public string Name
        {
            get
            {
                return name;
            }
            set
            {
                name = value.ToUpper();
            }
        }
        private int id;
        public int Id { get => id; set => id = value; }
        private DateTime? birthDay;
        public DateTime? BirthDay
        {
            get
            {
                return birthDay;
            }
            set
            {
                birthDay = value;
            }
		}
        // Propiedad autoimplementada
        public String Nationality { get; set; }

        private List<Account> accounts;
        public void AddAccount(Account account)
        {
            accounts.Add(account);
        }
        public List<Account> GetAccounts()
        {
            return accounts;
        }
        public void RemoveAccount(Account account)
        {
            accounts.Remove(account);
        }
        public override string ToString()
        {
            return name;
        }
    }
}

File: EntitiesVb/Customer.vb
Imports System.Threading

Public Class Customer
    Public Sub New()
        Nationality = "Argentino"
    End Sub
    Public Sub New(name As String, id As Integer)
        Me.New()
        Me.Name = name
        Me.Id = id
    End Sub
    Public Sub New(name As String, id As Integer, birthDay As Date)
        Me.New(name, id)
        Me.BirthDay = birthDay
    End Sub
    ' Campo, este forma parte de la implementación privada
    Private _name As String
    ' Propiedad, forma parte de la interfaz pública
    Public Property Name As String
        ' Metodo Getter es para obtener el valor
        Get
            Return _name
        End Get
        ' Método Setter es para asignar el valor
        Set(value As String)
            ' aqui se pueden hacer validaciones
            _name = value.ToUpper()
        End Set
    End Property

    Private _id As Integer
    Public Property Id As Integer
        Get
            Return _id
        End Get
        Set(value As Integer)
            _id = value
        End Set
    End Property

    Private _birthDay As Date
    Public Property BirthDay As Date
        Get
            Return _birthDay
        End Get
        Set(value As Date)
            _birthDay = value
        End Set
    End Property
    ' Propiedad autoimplementada
    Public Property Nationality As String

    Private _accounts As List(Of Account)
    Public Sub AddAccount(account As Account)
        _accounts.Add(account)
    End Sub
    Public Function GetAccounts() As List(Of Account)
        Return _accounts
    End Function
    Public Sub RemoveAccount(account As Account)
        _accounts.Remove(account)
    End Sub
    Public Overrides Function ToString() As String
        Return Name
    End Function

End Class
